src/features/home/components/comment-detail-posting.components.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import styled from "styled-components/native";
+import { Avatar, IconButton, Card, TouchableRipple } from "react-native-paper";
+import { Text } from "../../../components/typography/text.components";
+import { colors } from "../../../infrastructure/theme/colors";
+
+const LikesNumberText = styled(Text)`
+  font-family: ${(props) => props.theme.fonts.body_600};
+  font-size: ${(props) => props.theme.fontSizes.caption};
+`;
+
+const NameText = styled(Text)`
+  font-family: ${(props) => props.theme.fonts.body_700};
+  font-size: ${(props) => props.theme.fontSizes.caption};
+`;
+
+const CommentSection = styled.View`
+  flex-direction: row;
+  padding-right: ${(props) => props.theme.space[3]};
+  padding-left: ${(props) => props.theme.space[3]};
+`;
+
+const RightSideCommentSection = styled.View`
+  flex: 1;
+  padding-left: ${(props) => props.theme.space[2]};
+`;
+
+const NameAndCommentCard = styled(Card)`
+  padding: ${(props) => props.theme.space[2]};
+  background-color: #f9f9f9;
+`;
+
+const BottomRightSideCommentSection = styled.View`
+  flex-direction: row;
+  align-items: center;
+  padding-left: ${(props) => props.theme.space[1]};
+  padding-bottom: ${(props) => props.theme.space[2]};
+`;
+
+const LikesSection = styled.View`
+  flex-direction: row;
+  align-items: center;
+  padding-left: ${(props) => props.theme.space[1]};
+`;
+
+export const CommentDetailPosting = ({ item, onNavigate }) => {
+  const [upvoted, setUpvoted] = useState(false);
+  const [downvoted, setDownvoted] = useState(false);
+
+  const clickLike = () => {
+    if (upvoted) {
+      setUpvoted(false);
+      setDownvoted(false);
+    } else {
+      setUpvoted(true);
+      setDownvoted(false);
+    }
+  };
+
+  const clickDislike = () => {
+    if (downvoted) {
+      setDownvoted(false);
+      setUpvoted(false);
+    } else {
+      setUpvoted(false);
+      setDownvoted(true);
+    }
+  };
```

Comment:
that sounds like you try to implement the duplicated code in order to inverse. I mean on the line codes 47-48 are both false. At the first time, that's alright. In the future, we need to address this in order to make improvements better.